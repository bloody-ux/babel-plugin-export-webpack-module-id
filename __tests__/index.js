/* eslint-disable no-template-curly-in-string */

const pluginTester = require('babel-plugin-tester')
const createBabylonOptions = require('babylon-options')
const plugin = require('../index')

const babelOptions = {
  filename: 'currentFile.js',
  parserOpts: createBabylonOptions({
    stage: 2
  })
}

pluginTester({
  plugin,
  babelOptions,
  snapshot: true,
  tests: {
    'export default function': `
      export default function() {
        var x = 5;
        console.log('xxxx');
      }
    `,
    'export default class': `
     export default class XXX{
        a() {

        }
      }
    `,
    'export default variable expression': `
      export default k = 12;
    `,
    'export default class variable': `
      class X {}
      export default X;
    `,
    'export default function variable': `
      function X() {}
      export default X;
    `,
    'export call expression': `
      export default connect()(function(){})
    `,
    'export with babel options': {
      code: `
      export default connect()(function(){})
      `,
      pluginOptions: {
        webpackModuleId: 'moduleId'
      }
    },
    'export with filename match': {
      code: `
        export default connect()(function(){})
      `,
      pluginOptions: {
        include: '^currentFile'
      }
    },
    'export with filename not match': {
      code: `
        export default connect()(function(){})
      `,
      pluginOptions: {
        include: '^currentFile$'
      }
    },
    'export with filename excluded successfully': {
      code: `
        export default connect()(function(){})
      `,
      pluginOptions: {
        exclude: '^currentFile'
      }
    },
    'export with filename excluded not match': {
      code: `
        export default connect()(function(){})
      `,
      pluginOptions: {
        exclude: '^currentFile$'
      }
    }
  }
})

