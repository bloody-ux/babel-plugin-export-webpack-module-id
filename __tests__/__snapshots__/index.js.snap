// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`export call expression 1`] = `
"
export default connect()(function(){})

      ↓ ↓ ↓ ↓ ↓ ↓

var _ref = connect()(function () {});

_ref.webpackModuleId = module.id;
export default _ref;
export const webpackModuleId = module.id;
"
`;

exports[`export default class 1`] = `
"
export default class XXX{
   a() {

   }
 }

      ↓ ↓ ↓ ↓ ↓ ↓

var _XXX = class XXX {
  a() {}

};

_XXX.webpackModuleId = module.id;
export default _XXX;
export const webpackModuleId = module.id;
"
`;

exports[`export default class variable 1`] = `
"
class X {}
export default X;

      ↓ ↓ ↓ ↓ ↓ ↓

class X {}

X.webpackModuleId = module.id;
export default X;
export const webpackModuleId = module.id;
"
`;

exports[`export default function 1`] = `
"
export default function() {
  var x = 5;
  console.log('xxxx');
}

      ↓ ↓ ↓ ↓ ↓ ↓

var _ref = function () {
  var x = 5;
  console.log('xxxx');
};

_ref.webpackModuleId = module.id;
export default _ref;
export const webpackModuleId = module.id;
"
`;

exports[`export default function variable 1`] = `
"
function X() {}
export default X;

      ↓ ↓ ↓ ↓ ↓ ↓

function X() {}

X.webpackModuleId = module.id;
export default X;
export const webpackModuleId = module.id;
"
`;

exports[`export default variable expression 1`] = `
"
export default k = 12;

      ↓ ↓ ↓ ↓ ↓ ↓

var _ref = k = 12;

_ref.webpackModuleId = module.id;
export default _ref;
export const webpackModuleId = module.id;
"
`;

exports[`export with babel options 1`] = `
"
export default connect()(function(){})

      ↓ ↓ ↓ ↓ ↓ ↓

var _ref = connect()(function () {});

_ref.moduleId = module.id;
export default _ref;
export const moduleId = module.id;
"
`;
